lib/tests/test_revset.rs

```diff
@@ -4162,6 +4163,30 @@ fn test_evaluate_expression_diff_contains() {
     );
 }
 
+#[test]
+fn test_evaluate_expression_diff_contains_non_utf8() {
```

Comment:
I got a bit confused here. Perhaps it'd help to replace the note that "haystack is usually a string" with something like the following to the commit description (if it's correct):

> We don't have a way (a good way?) to search for non-UTF-8 strings with `diff_contains`, but this allows the user to search for Unicode patterns
inside not-necessarily-Unicode files.

I think that if we want to allow searching for non-Unicode haystacks, we need to start supporting some sort of `\x81` syntax to express them in Unicode.

I guess another way is to rely on TOML's support for such syntax, but that would mean that our config values are no longer String-s (or are they not already)? On Windows, I'm guessing the command line is UTF-16 or something.

Reply:
Something similar could also be added to the test as a comment.

Reply:
Updated the commit description. This function merely test that a non-utf-8 line isn't silently ignored, so didn't add detailed explanation about unicode/non-unicode patterns.

You can use bytes regex to match `\x80`, etc. btw.